src/graph.py

```diff
@@ -581,6 +583,46 @@ def grid_search(self, prize_file, Ws, Bs, Gs):
         return self.grid_randomization(prize_file, Ws, Bs, Gs, 0, 0)
 
 
+    ###########################################################################
+                #######          Summarize randomization results          #######
+    ###########################################################################
+
+    def generate_basic_statistics(self, robust_results): 
+        """
+        Summarizes robust network randomization results
+
+        Arguments: 
+            robust_results (dict of networkx): networkx robust subgraphs indexed by paramstring
+
+        Returns: 
+            pd.DataFrame: summary of each robust network
+        """
+        robust_summary = {}
+
+        for paramstring, robust_network in robust_results.items(): 
+        
+            if robust_network.number_of_nodes() != 0: 
+                
+                robust_df = get_networkx_graph_as_dataframe_of_nodes(robust_network)
+                robust_summary[paramstring] = {
+                    "W": paramstring.split("_")[1],
+                    "B": paramstring.split("_")[3],
+                    "G": paramstring.split("_")[5],
+                    "size": len(robust_df), 
+                    "min_robustness":   robust_df.robustness.min(), 
+                    "mean_robustness":  robust_df.robustness.mean(), 
+                    "max_specificity":  robust_df.specificity.max(),
+                    "mean_specificity": robust_df.specificity.mean(), 
+                    "mean_log_degree": np.log2(robust_df.degree).mean(), 
+                    "std_log_degree": np.log2(robust_df.degree).std(), 
+                    "KS_statistic": stats.ks_2samp(np.log2(robust_df.degree), np.log2(self.node_attributes.degree))[0]
+                }
+                
+        robust_summary = pd.DataFrame.from_dict(robust_summary, orient='index')
+        robust_summary["euclidean_distance"] = ((1 - robust_summary.mean_robustness) ** 2 + robust_summary.mean_specificity ** 2) ** 0.5
```

Comment:
Leave a comment here explaining why you're doing this outside of the `for` loop. Also mention what the distance is "from" / "to" which is implicit in this line I think.

Reply:
Removed this feature since it's a bit arbitrary.